functional/adoptopenjdk/src/net/adoptopenjdk/test/ShenandoahTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.adoptopenjdk.test;
+
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import static net.adoptopenjdk.test.JdkPlatform.Architecture;
+import static net.adoptopenjdk.test.JdkPlatform.OperatingSystem;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Tests whether Shenandoah GC is available.
+ * <p/>
+ * Shenandoah GC was enabled by default with JDK 15 (JEP 379) and backported to 11.0.9.
+ *
+ * @see <a href="https://openjdk.java.net/jeps/379">JEP 379: Shenandoah: A Low-Pause-Time Garbage
+ * Collector (Production)</a>
+ * @see <a href="https://bugs.openjdk.java.net/browse/JDK-8250784">JDK-8250784 (Backport)</a>
+ * @see <a href="https://wiki.openjdk.java.net/display/shenandoah/Main#Main-SupportOverview">Shenandoah Support
+ * Overview</a>
+ */
+@Test(groups = {"level.extended"})
+public class ShenandoahTest {
```

Comment:
Are there other features that we might expect to check in the future that would/could also live in a CheckXXOptionAvailableTest?

Reply:
ZGC, JFR to begin with.

Reply:
Tests for both added.